plugins/scss/index.js

```diff
@@ -154,14 +159,54 @@ const writer = function(thisPackage, outputDir) {
       file = path.resolve(outputDir, 'combined.scss');
       args.push(file);
       console.log('Writing ' + file);
-      promises.push(concat(scssEntries, file));
+      promises.push(concat.concatAsync(scssEntries, file));
     }
 
     file = path.resolve(outputDir, 'node-sass-args');
     console.log('Writing ' + file);
     promises.push(fs.writeFileAsync(file, args.join(' ')));
 
-    return Promise.all(promises);
+    return new Promise(function(resolve, reject) {
+      resolve(Promise.all(promises).then(function() {
+        var promises = [];
+        // Write a file for each theme also
+        if (utils.isAppPackage(thisPackage) &&
+            thisPackage.build.scss &&
+            thisPackage.build.themes &&
+            thisPackage.build.themes.length) {
+          // Take off the combined.scss generic file
+          args.pop();
+
+          var themeDir = outputDir + '/themes';
+          mkdirp.sync(themeDir);
+
+          thisPackage.build.themes.push('default');
+
+          // For all the themes, create a combined.scss and a node-sass-args file
+          thisPackage.build.themes.forEach(function(theme) {
+            var themeOutput = path.resolve(themeDir, theme + '.combined.scss');
+            var themeArgs = path.resolve(themeDir, theme + '.node-sass-args');
+            promises.push(fs.writeFileAsync(themeArgs, args.join(' ') + ' ' + themeOutput));
+            promises.push(new Promise(function(resolve, reject) {
```

Comment:
Same here. There is no need to make a `new Promise()` wrapper when you could return the `fs.writeFileAsync()` promise directly.